xarray/core/concat.py

```diff
@@ -189,6 +190,21 @@ def process_subset_opt(opt, subset):
                 # all nonindexes that are not the same in each dataset
                 for k in getattr(datasets[0], subset):
                     if k not in concat_over:
+                        equals[k] = None
+                        variables = [ds.variables[k] for ds in datasets]
+                        # first check without comparing values i.e. no computes
+                        for var in variables[1:]:
+                            equals[k] = getattr(variables[0], compat)(
+                                var, equiv=lazy_array_equiv
+                            )
+                            if not equals[k]:
```

Comment:
same concern as below -- should this be checking against `None` explicitly, which I believe `lazy_array_equiv` can return?

Reply:
I think this is right. We are exiting early because `equals[k] is not True` i.e. either the shapes are not equal, or one (or both) of the arrays is a numpy array or the dask names are not equal